src/stream/src/executor/sink.rs

```diff
@@ -123,59 +220,21 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
             assert_eq!(sink_input_schema.data_types(), info.schema.data_types());
         }
 
-        let stream_key = info.pk_indices.clone();
-        let downstream_pk = sink_param.downstream_pk_or_empty();
-        let stream_key_sink_pk_mismatch = stream_key.iter().any(|i| !downstream_pk.contains(i));
```

Comment:
Originally, when the `stream_key` is not empty, `stream_key_sink_pk_mismatch` is the same as `!pk_specified_and_matched`.